### PR TITLE
fix forking for scala 2.11

### PR DESCRIPTION
**Pull request by danielnorberg**
Inspired by https://gist.github.com/retronym/0178c212e4bacffed568#lambdas-in-scala-212-or-211-with-experimental-options

### CODE REVIEW

File: flo-scala_2.12/src/test/scala/com/spotify/flo/DslTest.scala [hunk 1]
```diff
@@ -36,5 +39,27 @@ class DslTest extends FlatSpec with Matchers {
     exception should have message "Builder accessor used outside a defTask scope"
   }
 
+  it should "runInProcesses" in {
+    val foo = defTaskNamed("foo")
+      .process(() => {
+        val jvm = ManagementFactory.getRuntimeMXBean.getName
```
Comment (danielnorberg) [inline flo-scala_2.12/src/test/scala/com/spotify/flo/DslTest.scala]: Surprisingly, two processes run but this code seems to be executed in the `bar` process. The `foo` process seems to return a closure, which is then evaluated in the `bar` process.

😕

Comment (danielnorberg) [reply flo-scala_2.12/src/test/scala/com/spotify/flo/DslTest.scala]: Maybe I'm misunderstanding the syntax here...

Comment (danielnorberg) [reply flo-scala_2.12/src/test/scala/com/spotify/flo/DslTest.scala]: nvm my bad, I had obviously declared my process fn to return a closure :shrug:

File: flo-scala_2.11/pom.xml [hunk 2]
```diff
@@ -53,7 +63,11 @@
           <sourceDir>../flo-scala_2.12/src/main/scala</sourceDir>
           <scalaVersion>${scala.version}</scalaVersion>
           <sendJavaToScalac>false</sendJavaToScalac>
-          <addScalacArgs>-Xexperimental</addScalacArgs>
+          <args>
+            <arg>-Xexperimental</arg>
+            <arg>-Ydelambdafy:method</arg>
```
Comment (honnix) [inline flo-scala_2.11/pom.xml]: `delambdafy`, lol

File: flo-scala_2.12/src/test/scala/com/spotify/flo/DslTest.scala [hunk 3]
```diff
@@ -36,5 +39,39 @@ class DslTest extends FlatSpec with Matchers {
...
+      .process({
+        val fooJvm = ManagementFactory.getRuntimeMXBean.getName
+        println(s"foo jvm: $fooJvm")
+        fooJvm
+      })
+    val bar: Task[Array[String]] = defTaskNamed("bar")
+      .input(foo)
+      .context(new TaskContextStrict[String, Array[String]] {
+        override def provide(evalContext: EvalContext): String = {
+          val barContextJvm = ManagementFactory.getRuntimeMXBean.getName
```
Comment (honnix) [inline flo-scala_2.12/src/test/scala/com/spotify/flo/DslTest.scala]: ⭐️

**Comment by danielnorberg** [comment]
I don't entirely understand the root cause of the issue that this resolves though. This just forces a quite different lambda serialization mechanism to be used.

**Comment by danielnorberg** [comment]
We might also need chill-scala if users pass scala things between tasks: https://github.com/twitter/chill#serializers-for-scala-classes

**Comment by danielnorberg** [comment]
https://github.com/spotify/flo/pull/94 might be preferable instead of using experimental compiler flags.

**Comment by danielnorberg** [comment]
Closing in favor of #94

**Pull request closed without merging**